articles/101_build_tool.md

```diff
@@ -0,0 +1,213 @@
+---
+layout: default
+title: A universal build tool
+permalink: articles/build_tool.html
+abstract:
+  This article describes the rationale for a universal build tool.
+published: true
+author: '[Dirk Thomas](https://github.com/dirk-thomas)'
+categories: Overview
+---
+
+{:toc}
+
+# {{ page.title }}
+
+<div class="abstract" markdown="1">
+{{ page.abstract }}
+</div>
+
+Original Author: {{ page.author }}
+
+## Preface
+
+ROS is developed with a federated model.
+It consists of many separately developed and maintained packages.
+
+The "manual" approach to build a set of packages consists of building all packages in their topological order one by one.
+For each package the documentation usually describes what the dependencies are, how to setup the environment to build the package as well as how to setup the environment afterwards to use the package.
+From an efficiency point of view that manual process does not scale well.
+
+This article describes the steps from the current build tools used in the ROS ecosystem to a single universal build tool which automates the process and works for various different use cases.
+
+## Goal
+
+The goal of the build tool is to build a set of packages with a single invocation automating the process.
+It should work with ROS 1 packages as well as ROS 2 packages which provide the necessary information in their manifest files.
+It should also work with packages that do not provide manifest files themselves, given that the meta information is externally provided.
+This will allow the build tool to be utilized for non-ROS dependencies of ROS packages (e.g. Gazebo including its dependencies).
+
+### Out of Scope
+
+The build tool does not cover the steps necessary to fetch the sources of the to-be-built packages.
+This can be done with existing tools (e.g. [rosinstall_generator](http://wiki.ros.org/rosinstall_generator), [vcstool](https://github.com/dirk-thomas/vcstool)).
```

Comment:
Please go ahead and create a PR for any changes you suggest. It makes it much easier to iterate on it and integrate the change. Thanks!